init_process.go

```diff
@@ -0,0 +1,406 @@
+		log.Panicln("Unable to get ECS Metadata")
+	}
+
+	res, err := http.Get(ecsTaskMetadataEndpointV4 + "/task")
+	if err != nil {
+		log.Panicln("failed to get ECS Metadata via HTTP Get")
+	}
+
+	response, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Panicln("failed to read response")
+	}
+	res.Body.Close()
+
+	err = json.Unmarshal(response, &metadata)
+	if err != nil {
+		log.Println(err)
+		log.Panicln("failed to unmarshal ECS metadata")
+	}
+
+	ARN, err := arn.Parse(metadata.ECS_TASK_ARN)
+	if err != nil {
+		log.Println(err)
+		log.Panicln("failed to parse ECS TaskARN")
+	}
+
+	resourceID := strings.Split(ARN.Resource, "/")
+	taskID := resourceID[len(resourceID)-1]
+	metadata.ECS_TASK_ID = taskID
+	metadata.AWS_REGION = ARN.Region
+	metadata.ECS_TASK_DEFINITION = metadata.ECS_FAMILY + ":" + metadata.ECS_REVISION
+
+	return metadata
+}
+
+// set ECS Task Metadata as environment variables in the invoker.sh
+func setECSTaskMetadataAsEnvVar(metadata ECSTaskMetadata, filePath string) {
+	t := reflect.TypeOf(metadata)
+	v := reflect.ValueOf(metadata)
+
+	initProcessEntrypointFile, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Unable to read init process entrypoint file: %s\n", filePath)
+	}
+	defer initProcessEntrypointFile.Close()
+
+	for i := 0; i < t.NumField(); i++ {
+		if v.Field(i).Interface().(string) == "" {
+			continue
+		}
+		writeContent := "export " + t.Field(i).Name + "=" + v.Field(i).Interface().(string) + "\n"
+		_, err = initProcessEntrypointFile.WriteString(writeContent)
+		if err != nil {
+			log.Println(err)
+			log.Fatalf("Cannot write %s in init process entrypoint file\n", writeContent[:len(writeContent)-2])
+		}
+	}
+}
+
+// create a new main config file
+func createMainConfigFile(filePath string) {
+	mainConfigFile, err := os.Create(filePath)
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot create main config file")
+	}
+	defer mainConfigFile.Close()
+}
+
+// add @INCLUDE original main config file to the new main config file
+func includeOriginalMainConfigFile(mainConfigFilePath, originalMainConfigFilePath string) {
+	mainConfigFile, err := os.OpenFile(mainConfigFilePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Unable to read main config file: %s\n", mainConfigFilePath)
+	}
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + originalMainConfigFilePath + "\n"
+	_, err = mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+// create Fluent Bit command to use new main config file
+func createCommand(command *string, filePath string) {
+	*command = *command + " -c /" + filePath
+}
+
+// create a folder to store S3 config files user specified
+func createS3ConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, os.ModePerm)
```

Comment:
I am wondering if there is a way to create a function to create either a file or a path based on the string? I think it should..

Reply:
If it's only for the init process, I can achieve this by determining if there is a` . ` in the file path. But I don't think that's very safe..  So I ended up setting create file and create folder as two functions